src/SimpleRADOSStriper.cc

```diff
@@ -488,12 +488,13 @@ ssize_t SimpleRADOSStriper::read(void* data, size_t len, uint64_t off)
   }
 
   size_t r = 0;
-  std::vector<std::pair<bufferlist, aiocompletionptr>> reads;
+  std::vector<std::pair<blptr, aiocompletionptr>> reads;
   while ((len-r) > 0) {
     auto ext = get_next_extent(off+r, len-r);
     auto& [bl, aiocp] = reads.emplace_back();
     aiocp = aiocompletionptr(librados::Rados::aio_create_completion());
-    if (int rc = ioctx.aio_read(ext.soid, aiocp.get(), &bl, ext.len, ext.off); rc < 0) {
+    bl = std::make_unique<librados::bufferlist>();
```

Comment:
the order of emplacing vs. 'filling in with values' is a bit convoluted.
Why not just emplacing the results of the get_next_extent() & the completion pointer?

Reply:
I did not want to change anything not related to the issue described in this commit. I agree that this order may be confusing at first sight. (and might be worth changing after resolving the segfault)

Reply:
I would do it here, as you are changing this specific code piece.
Your change is what would make it that extra bit of confusing.

Reply:
c0de3390a2a8907fa136d4cc1a1800452bbd58b0